app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java

```diff
@@ -233,17 +233,28 @@ public void onChanged(@Nullable String url) {
         setBlockingEnabled(session.isBlockingEnabled());
 
         session.getLoading().observe(this, new NonNullObserver<Boolean>() {
+            private long startLoadTime = 0;
+
             @Override
             public void onValueChanged(@NonNull Boolean loading) {
                 if (loading) {
+                    Log.i(FRAGMENT_TAG, "Starting to load page at " + getUrl());
+                    startLoadTime = SystemClock.elapsedRealtime();
+                    Log.i(FRAGMENT_TAG, "zerdatime " + startLoadTime +
+                            " - page load start");
+                    TelemetryWrapper.startLoad(startLoadTime);
+
                     backgroundTransitionGroup.resetTransition();
 
                     progressView.setProgress(5);
                     progressView.setVisibility(View.VISIBLE);
-                    TelemetryWrapper.startLoad(SystemClock.elapsedRealtime());
                 } else {
                     if (progressView.getVisibility() == View.VISIBLE) {
-                        TelemetryWrapper.endLoad(SystemClock.elapsedRealtime());
+                        long endTime = SystemClock.elapsedRealtime();
+                        Log.i(FRAGMENT_TAG, "zerdatime " + endTime +
+                                " - page load stop");
+                        Log.i(FRAGMENT_TAG, (endTime - startLoadTime) + " - elapsed load");
+                        TelemetryWrapper.endLoad(endTime);
```

Comment:
As all this is independent from the other code in the observer and also independent from BrowserFragment, we could move this code in a separate observer and even into a different file.